### PR TITLE
[Merged by Bors] - refactor: make `LinearMap.IsAdjointPair` accept bare functions as arguments

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by j-loreaux** [comment]
@ocfnash I didn't touch `LinearMap.BilinForm.IsAdjointPair` at all here. In part, I was confused by the docstring on the `LinearAlgebra.SesquilinearForm` file which says:

> These forms are a special case of the bilinear maps defined in BilinearMap.lean and all basic lemmas about construction and elementary calculations are found there.

I don't see how these are a special case. If you want me to do something with `BilinForm.IsAdjointPair` I can do that too, just let me know what you think is best.

**Comment by ocfnash** [comment]
> @ocfnash I didn't touch `LinearMap.BilinForm.IsAdjointPair` at all here. In part, I was confused by the docstring on the `LinearAlgebra.SesquilinearForm` file which says:
> 
> > These forms are a special case of the bilinear maps defined in BilinearMap.lean and all basic lemmas about construction and elementary calculations are found there.

I think this doc string should be excised. I believe it's trying to express that much of the API for bilinear forms (values in scalars) is a special case of that for bilinear maps (values in any module) but I think it is confusing and prone to become a false statement.

> I don't see how these are a special case. If you want me to do something with `BilinForm.IsAdjointPair` I can do that too, just let me know what you think is best.

Could you try deleting it? I think it might be almost unused.

**Comment by j-loreaux** [comment]
@ocfnash indeed, they were virtually unused. I've deleted them in this PR, but if you would prefer that to be a separate PR, it's just a `git revert` and `git cherry-pick` away.

File: Mathlib/LinearAlgebra/SesquilinearForm.lean [hunk 1]
```diff
@@ -15,8 +15,8 @@ form `M₁ →ₛₗ[I₁] M₂ →ₛₗ[I₂] M`, where `I₁ : R₁ →+* R`
 Sesquilinear forms are the special case that `M₁ = M₂`, `M = R₁ = R₂ = R`, and `I₁ = RingHom.id R`.
 Taking additionally `I₂ = RingHom.id R`, then one obtains bilinear forms.
 
-These forms are a special case of the bilinear maps defined in `BilinearMap.lean` and all basic
-lemmas about construction and elementary calculations are found there.
+Sesquilinear maps are a special case of the bilinear maps defined in `BilinearMap.lean` and all
```
Comment (ocfnash) [inline Mathlib/LinearAlgebra/SesquilinearForm.lean:18]:
```suggestion
Sesquilinear maps are a special case of the bilinear maps defined in `BilinearMap.lean` and `many`
```

**Review by ocfnash — APPROVED** [review]
Thanks!

**Comment by ocfnash** [comment]
Aside from the headline change, it's great to see the old duplicated `LinearMap.BilinForm.IsAdjointPair` etc killed off.

I don't suppose I can tempt you to consider the less trivial task of killing off [LinearMap.BilinForm.Nondegenerate](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/BilinearForm/Properties.html#LinearMap.BilinForm.Nondegenerate) since we have [LinearMap.Nondegenerate](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/SesquilinearForm.html#LinearMap.Nondegenerate)

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**